### ISSUE TITLE
Support for categorical/ordinal data?

### ISSUE THREAD

**Issue by buhrmann** [minepy/minepy#23]
Hi, I'm currently evaluating different measures to detect dependencies in high-dimensional data. From what I've read about MIC/TIC these are currently only defined, and here implemented, for quantitative variables, is that correct? Are you working on any ideas to deal with categorical data, or do you know of any workaround/transformation to make that work?

**Comment by davidealbanese** [comment]
Dear @buhrmann,
for categorical data I suggest to use the (normalized) mutual information. For a Python implementation see [sklearn.metrics.normalized_mutual_info_score](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.mutual_info_score.html#sklearn.metrics.mutual_info_score). On the other hand, MIC/TIC can handle ordered variables, since they are rank-based methods. Simply pass integer numbers to the function. The algorithm never splits contiguous values of the same category during the optimization:

```python
>>> import minepy
>>> x = [1,1,1,2,2,2,3,3,3,2,2,2,1,1,1]
>>> y = [1,1,1,2,2,2,3,3,3,4,4,4,5,5,5]
>>> mine = minepy.MINE(alpha=0.8)
>>> mine.compute_score(x, y)
>>> mine.mic()
0.9709505944546685
>>> x2 = [1,2,1,2,1,2,1,2,1,2,1,2,1,2,1]
>>> mine.compute_score(x2, y)
>>> mine.mic()
0.04581413154894311
```

Davide

**Comment by buhrmann** [comment]
Hi, thanks for the quick response! My concern with mixing MI and MIC/TIC is whether these statistics are comparable, at least between variables in the same dataset. I.e., would it make sense to calculate MI for some variable pairs and MIC/TIC for others, and then rank the dependencies on a single scale? Also, it would seem then that for both mixed-type MI (continuous-categorical) and categorical-categorical I would have to resort to the kind of approximations implemented in e.g. in R's mpmi (or NPEET in Python), right?

**Comment by davidealbanese** [comment]
Obviously you can not mix MI and MIC/TIC measures, as they are not comparable. Honestly, I don't have any experience with mpmi and NPEET libraries, but, from what I understand, they implement algorithms for estimating entropy and MI from continuous and categorical variables.

Davide

**Comment by buhrmann** [comment]
Ok. Thanks for the clarification!
